erts/emulator/beam/erl_db_util.c

```diff
@@ -2338,8 +2341,8 @@ Eterm db_prog_match(Process *c_p,
             *esp++ = t;
             break;
 	case matchCall0:
-	    bif = (Eterm (*)(Process*, ...)) *pc++;
-	    t = (*bif)(build_proc, bif_args);
+	    bif_call_0 = (BIF_RETTYPE (*)(BIF_ALIST_0)) *pc++;
+	    t = (*bif_call_0)(build_proc, esp-1, bif_args);
```

Comment:
bif_args should be the second parameter, not the third.  The third parameter is normally the beam instruction pointer, which most bifs won't need, I _think_ it might work to just pass NULL there.

Reply:
Yes, passing in NULL will work. None of the BIFs that can actually be called will use it.